src/Symfony/Component/Cache/Adapter/PdoAdapter.php

```diff
@@ -108,9 +108,20 @@ public function createTable()
         $conn = $this->getConnection();
 
         if ($conn instanceof Connection) {
+            $types = array(
+                'mysql'  => 'binary',
+                'sqlite' => 'text',
+                'pgsql'  => 'string',
+                'oci'    => 'string',
+                'sqlsrv' => 'string',
+            );
+            if (!isset($types[$this->driver])) {
+                throw new \DomainException(sprintf('Creating the cache table is currently not implemented for PDO driver "%s".', $this->driver));
+            }
+            
             $schema = new Schema();
             $table = $schema->createTable($this->table);
-            $table->addColumn($this->idCol, 'blob', array('length' => 255));
+            $table->addColumn($this->idCol, $types[$this->driver], array('length' => 255));
```

Comment:
I indeed mentioned the MySQL case explicitly. However, [the section below](https://github.com/symfony/symfony/blob/f8b02ed038cb560298f52094b62bd18d474cad28/src/Symfony/Component/Cache/Adapter/PdoAdapter.php#L126-L149) differs from the schema, in the sense that none of the statements there define the id column as blob (expecting you mean blob, not blog). Mapping it like this should result in an id column type that matches up with those below.